### PR TITLE
Dynamically initialize posthog

### PR DESCRIPTION
**Pull request by Mastersam07**
## :bulb: Motivation and Context
Posthog initialization can only be done natively from android manifest and info.plist. However, there are cases where user want to configure dynamically from the dart side.

A good example is loading the posthog api key and host from some 3rd party(say api or firebase remote config) and then setting up posthog.

The current setup does not allow

This PR aims at  solving the above described issue.

## :green_heart: How did you test it?
With the example code, on an emulator.

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [x] I updated the docs if needed.
- [x] No breaking change or entry added to the changelog.

### CODE REVIEW

**Comment by marandaneto** [comment]
@Mastersam07 sorry I am out on sick leave and will come back to it either later this week or next week.

**Comment by GowthamanRavichandran3** [comment]
Hi @marandaneto,  This PR will also fulfill our requirements. So, please merge this PR as soon as possible.

Thanks,
Gowtham.

File: android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt [hunk 1]
```diff
@@ -56,86 +48,56 @@ class PosthogFlutterPlugin : FlutterPlugin, MethodCallHandler {
             }
             PostHogAndroid.setup(applicationContext, config)
 
-        } catch (e: Throwable) {
```
Comment (marandaneto) [inline android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt:59]: This should not change, `Throwable` is lower level

File: android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt [hunk 2]
```diff
@@ -15,37 +15,29 @@ import io.flutter.plugin.common.MethodChannel.Result
...
-
+        applicationContext = flutterPluginBinding.applicationContext
+        initPlugin(applicationContext)
         channel.setMethodCallHandler(this)
     }
 
-    private fun initPlugin(applicationContext: Context) {
+    private fun initPlugin(context: Context) {
         try {
-            // TODO: replace deprecated method API 33
```
Comment (marandaneto) [inline android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt:34]: Please add the comment back

File: android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt [hunk 3]
```diff
@@ -15,37 +15,29 @@ import io.flutter.plugin.common.MethodChannel.Result
 
 /** PosthogFlutterPlugin */
 class PosthogFlutterPlugin : FlutterPlugin, MethodCallHandler {
-    /// The MethodChannel that will the communication between Flutter and native Android
```
Comment (marandaneto) [inline android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt:18]: Some comments were removed, please add them back, those comments are good for new comers

File: ios/Classes/PosthogFlutterPlugin.swift [hunk 4]
```diff
@@ -20,36 +20,33 @@ public class PosthogFlutterPlugin: NSObject, FlutterPlugin {
...
         let apiKey = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.API_KEY") as? String ?? ""
+        let host = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.POSTHOG_HOST") as? String ?? PostHogConfig.defaultHost
+        let trackLifecycleEvents = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.CAPTURE_APPLICATION_LIFECYCLE_EVENTS") as? Bool ?? false
+        let debug = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.DEBUG") as? Bool ?? false
 
-        if apiKey.isEmpty {
-            print("[PostHog] com.posthog.posthog.API_KEY is missing!")
-            return
-        }
+        print("\nApiKey:", apiKey)
```
Comment (marandaneto) [inline ios/Classes/PosthogFlutterPlugin.swift:29]: prints should be removed

**Comment by dharanidharandharmasivam** [comment]
@marandaneto , can you please provide some approximate timeline to merge this changes to merge,  since am having scenario like based on domain I needs to log in different things, so can you please merge this PR sooner

**Comment by marandaneto** [comment]
> @marandaneto , can you please provide some approximate timeline to merge this changes to merge, since am having scenario like based on domain I needs to log in different things, so can you please merge this PR sooner

I left a few comments and specifically a different approach for compatibility, described [here](https://github.com/PostHog/posthog-flutter/pull/113#issuecomment-2288733844), would you apply those suggestions so we can move forward?

**Comment by Mastersam07** [comment]
> > @marandaneto , can you please provide some approximate timeline to merge this changes to merge, since am having scenario like based on domain I needs to log in different things, so can you please merge this PR sooner
> 
> I left a few comments and specifically a different approach for compatibility, described [here](https://github.com/PostHog/posthog-flutter/pull/113#issuecomment-2288733844), would you apply those suggestions so we can move forward?

I will attend to these changea. I was on leave.

**Comment by marandaneto** [comment]
Closed in favor of https://github.com/PostHog/posthog-flutter/pull/117
Thanks for the initial PR @Mastersam07

**Pull request closed without merging**